src/util/Retry.hpp

```diff
@@ -100,15 +107,18 @@ class Retry {
     void
     retry(Fn&& func)
     {
+        *canceled_ = false;
         timer_.expires_after(strategy_->getDelay());
         strategy_->increaseDelay();
-        timer_.async_wait([this, func = std::forward<Fn>(func)](boost::system::error_code const& ec) {
-            if (ec == boost::asio::error::operation_aborted) {
-                return;
+        timer_.async_wait(
+            [this, canceled = canceled_, func = std::forward<Fn>(func)](boost::system::error_code const& ec) {
+                if (ec == boost::asio::error::operation_aborted or *canceled) {
```

Comment:
The error check is different here https://github.com/XRPLF/clio/blob/develop/src/util/requests/impl/WsConnectionImpl.hpp#L137. 
Not sure which one is correct.

Reply:
It is `if (...) do_something();` there and here it is `if (...) return;` that's why they are different.